snippets/array-remove.md

```diff
@@ -1,13 +1,22 @@
 ### Array remove
 
-Use `Array.filter()` to find array elements that return truthy values and `Array.reduce()` to remove elements using `Array.splice()`.
-The `func` is invoked with three arguments (`value, index, array`).
+Finds elements that satisfy a predicate and returns them, at the same time modifying an original array.
+In other words, it works like [R#partition](http://ramdajs.com/docs/#partition), except that it mutates the original array and returns another partition instead of returning two.
 
 ```js
-const remove = (arr, func) =>
-  Array.isArray(arr) ? arr.filter(func).reduce((acc, val) => {
-    arr.splice(arr.indexOf(val), 1); return acc.concat(val);
-    }, [])
-  : [];
-// remove([1, 2, 3, 4], n => n % 2 == 0) -> [2, 4]
+/**
+ * @param {E[]} arr
+ * @param {(E, number, E[]) => boolean}
```

Comment:
Refer to the styleguide (CONTRIBUTING.md) for this. We do not write function signatures in snippets.